### PR TITLE
Copter: Add option to resume precland after reposiiton

### PR DESCRIPTION
**Pull request by rishabsingh3003**
Currently, if the user were to reposition the vehicle during a Prec Land manually, it would cancel Prec Land and not resume it even if the target was in view (i.e Copter just does a normal land). 
By using PLND_OPTIONS parameter, one can now allow resuming prec land even after a manual reposition. 

This has been tested on SITL

### CODE REVIEW

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -668,6 +668,10 @@ void Mode::land_run_horizontal_control()
                     AP::logger().Write_Event(LogEvent::LAND_REPO_ACTIVE);
                 }
                 copter.ap.land_repo_active = true;
+            } else {
+                if (copter.ap.land_repo_active && copter.precland.allow_precland_after_reposition()) {
```
Comment (khancyr) [inline ArduCopter/mode.cpp:672]: Copter.ap.land_repo_active is totally redundant here.

Comment (khancyr) [inline ArduCopter/mode.cpp:672]: Probably need a guard to check that precland is available otherwise the copter.precland object won't exist

**Review by khancyr — CHANGES_REQUESTED** [review]
Need a #if guard again precland to work.

Otherwise, the feature can be interesting and the implementation won't impact much the code if the option isn't enabled. So that is fine.

**Comment by rishabsingh3003** [comment]
@khancyr thanks! I have fixed those things, however, the RTL land_repo_active isn't related to precland in any way. For prec land, RTL calls the same controller, which we have already fixed in this PR.

**Comment by khancyr** [comment]
yep, my bad on this, was looking at the wrong codebase. that is fine then as indeed descent_run() don't use precland yet

**Comment by rmackay9** [comment]
Merged thanks!

**Comment by rmackay9** [comment]
This is included in Copter-4.3.4-rc1

**Pull request merged**